src/Arachne.cc

```diff
@@ -1118,22 +1131,20 @@ init(int* argcp, const char** argv) {
     while (numActiveCores != minNumCores)
         usleep(1);
 
+    mainThreadInit();
     // Only consider Arachne initialized if we've successfully parsed options
     // allocated all resources, and connected to the CoreArbiter
     initialized = true;
 }
 
 /**
- * This function should be invoked in unit test setup to make Arachne
- * functions callable from the unit test, which is not an Arachne thread.
- *
- * This function sets up just enough state to allow the current thread to
- * execute unit tests which call Arachne functions.
- * We assume that the unit tests are run from the main kernel thread which
- * will never swap out when running the dispatch() loop.
+ * Set up just enough state to allow the current thread to invoke Arachne
+ * functions, including createThreadOnCore. It is automatically invoked in the
+ * caller of Arachne::init(), which is not running on an Arachne core. This
+ * solves the bootstrapping problem.
```

Comment:
May need to mention that we put PerfStats allocation & initialization here.

Reply:
I've updated the comment.